react/components/Slider.js

```diff
@@ -487,26 +492,39 @@ class Slider extends PureComponent {
     }
   }
 
+  getSlidesToSlide = () => {
+    const { navigationStep, scrollByPage } = this.props
+
+    if (typeof navigationStep === 'number') {
+      return navigationStep
+    } else if (navigationStep === constants.NAVIGATION_PAGE || scrollByPage) {
+      return this.perPage
+    }
+
+    return 1
+  }
+
   renderArrows = () => {
     const {
       arrowsContainerComponent: ArrowsContainerComponent,
       arrowRender,
-      scrollByPage,
     } = this.props
 
     if (!arrowRender) {
       return null
     }
 
+    const slidesToSlide = this.getSlidesToSlide()
+
     const arrows = (
       <Fragment>
         {arrowRender({
           orientation: 'left',
-          onClick: scrollByPage ? this.prevPage : () => this.prev(),
+          onClick: () => this.prev(slidesToSlide),
```

Comment:
```suggestion
          onClick: () => this.prev(this.getSlidesToSlide),
```

Reply:
If I do this I'll be passing a function as an argument instead of a value

Reply:
understand, really.